industrialucn/optimization/test_econstraint.py

```diff
@@ -0,0 +1,290 @@
+import logging
+import sys
+import unittest
+
+import numpy as np
+
+from industrialucn.optimization.econstraint import all_weakly_non_dominated, all_non_dominated
+from . import econstraint as ec
+
+
+def config_logger(logger, level):
+    logger.setLevel(level)
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+
+config_logger(ec.logger, logging.DEBUG)
+
+# define parameters
+n = 15  # number of parks
+m = 50  # number of buildings
+rnd = np.random
+rnd.seed(0)
+cpx = rnd.rand(n) * 300  # x coordinate for parks
+cpy = rnd.rand(n) * 100  # y coordinate for parks
+cbx = rnd.rand(m) * 300  # x coordinate for buildings
+cby = rnd.rand(m) * 100  # y coordinate for buildings
+parks = [i for i in range(n)]
+buildings = [j for j in range(m)]
+distance = {(i, j): np.hypot(cbx[i] - cpx[j], cby[i] - cpy[j])
+            for i in buildings for j in parks}
+
+
+class EConstraint(unittest.TestCase):
+
+    def test_dominance(self):
+        v1 = [[1.0, 4324.8, 165.4],
+              [2.0, 2757.5, 116.1],
+              [3.0, 2106.9, 74.8],
+              [13.0, 1351.1, 56.9],
+              [2.0, 2757.5, 116.1],
+              [2.0, 2966.0, 107.7],
+              [3.0, 2106.9, 74.8],
+              [13.0, 1351.1, 56.9]]
+
+        v2 = [[1.0, 4324.8, 165.4],
+              [2.0, 2757.5, 116.1],
+              [3.0, 2106.9, 74.8],
+              [13.0, 1351.1, 56.9],
+              [2.0, 2757.5, 116.1],
+              [2.0, 2966.0, 107.7],
+              [3.0, 2106.9, 74.8],
+              [13.0, 1351.1, 56.9],
+              [3.0, 2223.9, 82.5]]
+
+        v3 = [[1.0, 4324.8, 165.4],
+              [2.0, 2757.5, 116.1],
+              [3.0, 2106.9, 74.8],
+              [13.0, 1351.1, 56.9],
+              [2.0, 2757.5, 116.1],
+              [2.0, 2966.0, 107.7],
+              [3.0, 2106.9, 74.8],
+              [13.0, 1351.1, 56.9],
+              [3.0, 2223.9, 82.5],
+              [2.0, 2106.9, 74.8]]
+
+        self.assertTrue(all_non_dominated(v1))
+        self.assertFalse(all_non_dominated(v2))
+        self.assertFalse(all_non_dominated(v3))
+
+        self.assertTrue(all_weakly_non_dominated(v1))
+        self.assertTrue(all_weakly_non_dominated(v2))
+        self.assertFalse(all_weakly_non_dominated(v3))
+
+    def test_cplex_mip(self):
```

Comment:
the name of the function needs more details

Reply:
Sure!
I made the change here 0b2563e96ded21f60427200f26f93a164b70849e